lib/label.ml

```diff
@@ -85,6 +85,7 @@ type t =
   | Set of string * string
   | Unset of string
   | Block_kind of block_kind
+  | Language_tag of string
```

Comment:
I agree and my first approach was to use `block_kind`/`syntax` for it, but the issue is that this doesn't preserve whether the language tag existed when it was parsed, thus it would add tags where the language was the same as automatically determined (creating diff noise) or if we implemented it to only emit it if it is not the default, it would remove the tags where the input had the default language (creating diff noise again). Thus I would argue that we *need* some place to track this.

It is true that `Language_tag` should only exist at most once, so a nicer way would be a `string option`. Though the same is true for `Block_kind`, and possibly other labels like `Version`. It seems `Label.t list` (as used in `Block.t`) is not a good way to express the labels, `Label.t` should probably be a record with `'a option` fields for many of the tags (in fact, only `Set`, `Unset` and `Env` seem like they should be lists).

Do you think it makes sense to refactor this before merging? I could do this, but part of the reason why this PR ended being so big is that I came across a lot of bits that needed drive-by refactoring so I was a bit hesistant on changing even more (especially as that might change the semantics and reject previously valid files for multiple tags). For practical reasons I think deferring this to a future PR might be better (partly for the fact that it would get a clear changelog entry).

Reply:
That's a good point, I added an explanation of the tag and a refactoring suggestion (and rebased to fix the changelog).